### PR TITLE
Refactors DefaultJavaPrettyPrinter and JDTTreeBuilder 🎉

### PR DESCRIPTION
**Pull request by GerardPaligot**

### CODE REVIEW

File: src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderQuery.java [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+						((FieldBinding) qualifiedNameReference.binding).declaringClass.compoundName);
+	}
+
+	/**
+	 * Checks if the last node in the stack in the context is an assignment and have a lhs equals to the given expression.
+	 *
+	 * @param context
+	 * 		Context of the {@link JDTTreeBuilder}.
+	 * @param lhs
+	 * 		Potential lhs of the assignment.
```
Comment (tdurieux) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderQuery.java]: What does `lhs` mean?

Comment (retuxx) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderQuery.java]: Usually, it means left hand side?

Comment (GerardPaligot) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderQuery.java]: Exactly.

Comment (GerardPaligot) [inline src/main/java/spoon/support/compiler/jdt/JDTTreeBuilderQuery.java]: For information, we already use rhs in our model.

File: src/main/java/spoon/reflect/declaration/CtTypeInformation.java [hunk 2]
```diff
@@ -86,6 +86,16 @@
...
 
 	/**
+	 * Returns true if this type is an annotation type.
+	 */
+	boolean isAnnotationType();
+
+	/**
+	 * Returns true if the types is a new class.
+	 */
+	boolean isNewClass();
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtTypeInformation.java]: isAnonymousClass?

File: src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java [hunk 3]
```diff
@@ -511,6 +511,26 @@ public boolean isInterface() {
...
+		} else {
+			return t.isAnnotationType();
+		}
+	}
+
+	@Override
+	public boolean isNewClass() {
+		CtType<T> t = getDeclaration();
+		if (t == null) {
+			return simplename == null || simplename.isEmpty();
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java]: test with runtime reflection with convention on simple name?

File: src/main/java/spoon/reflect/factory/CodeFactory.java [hunk 4]
```diff
@@ -367,7 +415,35 @@ public CodeFactory(Factory factory) {
...
 			va = factory.Core().createVariableRead();
 		}
-		return va.setVariable(variable).setType(variable.getType() == null ? null : variable.getType().clone());
+		return va.setVariable(variable);
+	}
+
+	/**
+	 * Creates variable access.
+	 */
+	public <T> CtVariableAccess<T> createVariableAccess(CtVariableReference<T> variableReference, boolean isReadAccess) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CodeFactory.java]: move to JDTHelpers

File: src/main/java/spoon/reflect/factory/CodeFactory.java [hunk 5]
```diff
@@ -367,7 +415,35 @@ public CodeFactory(Factory factory) {
...
+		} else {
+			variableAccess = factory.Core().createVariableRead();
+		}
+		return variableAccess.setVariable(variableReference);
+	}
+
+	/**
+	 * Creates field access.
+	 */
+	public <T> CtFieldAccess<T> createFieldAccess(CtVariableReference<T> variableReference, CtExpression<?> target, boolean isReadAccess) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CodeFactory.java]: move to JDTHelpers

**Comment by GerardPaligot** [comment]
@monperrus PR ok. :)

**Pull request merged**